presto-main/src/main/java/com/facebook/presto/resourcemanager/ResourceManagerClusterStateProvider.java

```diff
@@ -157,8 +160,12 @@ public void registerNodeHeartbeat(NodeStatus nodeStatus)
     }
 
     public List<ResourceGroupRuntimeInfo> getClusterResourceGroups(String excludingNode)
+            throws ResourceManagerInconsistentException
     {
         requireNonNull(excludingNode, "excludingNode is null");
+        if (!isConsistent()) {
+            throw new ResourceManagerInconsistentException();
```

Comment:
I found this exception being used by many places. Does it make sense to make it more verbose? Check `GenericInternalException` as an example. Or a more general question: is there anything preventing us using `PrestoException`?

Reply:
Not addressed? Or there is some rationale behind it?

Reply:
Added a descriptive error message that can be customzied.